synthesizer/snark/src/certificate/mod.rs

```diff
@@ -41,9 +43,11 @@ impl<N: Network> Certificate<N> {
 
         // Retrieve the proving parameters.
         let degree_info = proving_key.circuit.index_info.degree_info::<N::Field, varuna::VarunaHidingMode>();
-        let universal_prover = N::varuna_universal_prover(degree_info);
         let fiat_shamir = N::varuna_fs_parameters();
 
+        let mut universal_prover = universal_prover.write();
+        universal_prover.update(srs, degree_info)?;
```

Comment:
```suggestion
        universal_prover.write().update(srs, degree_info)?;
```

Reply:
unless it should not be read from or written to until the end of the method

Reply:
Good one 1d76c1aaa